components/external_storage/export/src/export.rs

```diff
@@ -180,7 +180,7 @@ fn create_backend_inner(
                 return Err(bad_backend(Backend::CloudDynamic(dyn_backend.clone())));
             }
         },
-        #[cfg(not(any(feature = "cloud-gcp", feature = "cloud-aws")))]
+        #[cfg(not(all(feature = "cloud-gcp", feature = "cloud-aws", feature = "cloud-azure")))]
```

Comment:
Why use all?

Reply:
We can omit the catch-all match arm only if all the features are provided. Otherwise, a non exhausted error will occur.

Reply:
Why not leave the non-exhausted branch?

Reply:
I don't know why this is in the first place. 😂 
According to the blame this `cfg` is introduced by @gregwebs in 70c7467c, and I believe the original line does not work as expected. (It will fail to compile when `cloud-gcp` is present but `cloud-aws` is not) This change in this PR is just a bugfix to make the code compile after https://github.com/pingcap/kvproto/commit/78a55fb8e69c1052a55945b69a3a571fd248bb7d breaks the code.

Reply:
Ok, so I suggest the following:
```suggestion
        #[allow(unreachable_patterns)]
```

Reply:
yes, this should have been `all`. Unreachable would make sense as well.